packages/frontend/src/utils/getVolunteers.ts

```diff
@@ -1,44 +1,48 @@
 import { FirebaseCollection } from 'uva-shared';
 import { firebaseDb } from 'src/main';
 import { IVolunteer } from 'uva-shared/src/types/volunteer';
-import { collection, getDocs, limit, orderBy, query, startAfter } from 'firebase/firestore';
+import { collection, getDocs, limit, orderBy, query, startAfter, where, DocumentSnapshot } from 'firebase/firestore';
 import { QueryArgs } from './getEvents';
 
 export interface GetVolunteersResult {
   readonly volunteers: IVolunteer[];
   readonly lastVolunteer: DocumentSnapshot | null;
 }
 
-import { DocumentSnapshot } from 'firebase/firestore';
-
 export const getVolunteers = async (
   name?: string,
   lastVolunteer?: DocumentSnapshot | null,
-  pageSize = 10,
+  pageSize = 20,
 ): Promise<GetVolunteersResult> => {
-  const queryArgs: QueryArgs[] = [orderBy('firstName', 'desc'), limit(pageSize)];
-
-  if (lastVolunteer) {
-    queryArgs.push(startAfter(lastVolunteer));
-  }
+  console.log(name, lastVolunteer, pageSize);
+  let volunteers: IVolunteer[] = [];
+  let querySnapshot;
+  
+  const searchByField = async (field: 'firstName' | 'lastName') => {
+    const queryArgs: QueryArgs[] = [orderBy(field, 'asc'), limit(pageSize)];
 
-  const queryForVolunteers = query(collection(firebaseDb, FirebaseCollection.Volunteers), ...queryArgs);
+    if (name) {
+      const start = name.toLowerCase();
+      const end = start + '\uf8ff';
+      queryArgs.unshift(where(field, '>=', start), where(field, '<', end));
+    }
 
-  try {
-    const querySnapshot = await getDocs(queryForVolunteers);
-    const volunteers = querySnapshot.docs.map(doc => doc.data() as IVolunteer);
+    if (lastVolunteer) {
+      queryArgs.push(startAfter(lastVolunteer));
+    }
 
-    if (name) {
-      const filteredVolunteers = volunteers.filter(
-        volunteer =>
-          volunteer.firstName.toLowerCase().includes(name.toLowerCase()) ||
-          volunteer.lastName.toLowerCase().includes(name.toLowerCase()),
-      );
+    const queryForVolunteers = query(collection(firebaseDb, FirebaseCollection.Volunteers), ...queryArgs);
+    return await getDocs(queryForVolunteers);
+  };
 
-      return {
-        volunteers: filteredVolunteers,
-        lastVolunteer: filteredVolunteers.length ? querySnapshot.docs[querySnapshot.docs.length - 1] : null,
-      };
+  try {
+    querySnapshot = await searchByField('firstName');
```

Comment:
Sorry for making more comments :'D

Let's make one query instead of two using or: https://firebase.google.com/docs/firestore/query-data/queries#or_queries, should be something like this:
```
or(
   and( where('firstName', '>=', start), where('firstName', '<', end) ),
   and( where('lastName', '>=', start), where('lastName', '<', end) ),
)
```

Reply:
That's ok :D
I have committed new version, which required to add some indexes in the Firebase, and after that, it works perfectly